arm/Microsoft.Authorization/roleDefinitions/deploy.bicep

```diff
@@ -1,81 +1,85 @@
+@sys.description('Optional. Location for all resources.')
 param location string = deployment().location
 
+@sys.description('Optional. Role definition assignable scopes. If not provided, will use the current scope provided.')
+param assignableScopes array = []
+
 module roleDefinition_mg '.bicep/nested_roleDefinitions_mg.bicep' = if (!empty(managementGroupId) && empty(subscriptionId) && empty(resourceGroupName)) {
-  name: 'roleDefinition-mg-${guid(roleName, managementGroupId, location)}'
+  name: '${uniqueString(deployment().name, location)}-roleDefinition-mg-Module'
   scope: managementGroup(managementGroupId)
   params: {
     roleName: roleName
-    roleDescription: roleDescription
+    description: description
     actions: actions
     notActions: notActions
-    dataActions: dataActions
-    notDataActions: notDataActions
     managementGroupId: managementGroupId
-    location: location
+    assignableScopes: assignableScopes
   }
 }
 
 module roleDefinition_sub '.bicep/nested_roleDefinitions_sub.bicep' = if (empty(managementGroupId) && !empty(subscriptionId) && empty(resourceGroupName)) {
-  name: 'roleDefinition-sub-${guid(roleName, subscriptionId, location)}'
+  name: '${uniqueString(deployment().name, location)}-roleDefinition-sub-Module'
   scope: subscription(subscriptionId)
   params: {
     roleName: roleName
-    roleDescription: roleDescription
+    description: description
     actions: actions
     notActions: notActions
     dataActions: dataActions
     notDataActions: notDataActions
     subscriptionId: subscriptionId
-    location: location
+    assignableScopes: assignableScopes
   }
 }
 
 module roleDefinition_rg '.bicep/nested_roleDefinitions_rg.bicep' = if (empty(managementGroupId) && !empty(resourceGroupName) && !empty(subscriptionId)) {
-  name: 'roleDefinition-rg-${guid(roleName, subscriptionId, resourceGroupName, location)}'
+  name: '${uniqueString(deployment().name, location)}-roleDefinition-rg-Module'
   scope: resourceGroup(subscriptionId, resourceGroupName)
   params: {
     roleName: roleName
-    roleDescription: roleDescription
+    description: description
     actions: actions
     notActions: notActions
     dataActions: dataActions
     notDataActions: notDataActions
     subscriptionId: subscriptionId
     resourceGroupName: resourceGroupName
-    location: location
+    assignableScopes: assignableScopes
   }
 }
 
+@sys.description('The GUID of the Role Definition')
 output roleDefinitionName string = !empty(managementGroupId) ? roleDefinition_mg.outputs.roleDefinitionName : (!empty(resourceGroupName) ? roleDefinition_rg.outputs.roleDefinitionName : roleDefinition_sub.outputs.roleDefinitionName)
-output roleDefinitionId string = !empty(managementGroupId) ? roleDefinition_mg.outputs.roleDefinitionId : (!empty(resourceGroupName) ? roleDefinition_rg.outputs.roleDefinitionId : roleDefinition_sub.outputs.roleDefinitionId)
+@sys.description('The Resource ID of the Role Definition')
```

Comment:
I'd suggest to add new lines between outputs and param

Reply:
added